### PR TITLE
Improve support for cases with empty --path provided to wp-cli commands

### PR DESCRIPTION
**Pull request by wojtekn**
I propose fixing the confusing messaging we show if an empty `--path` parameter is provided to command calls.

The empty parameter is considered a `true` flag, let's treat it as empty path:

```
% bin/wp import --path
Error: The --path parameter cannot be empty when it is provided
```

This is also an empty parameter considered a `true` flag because there is no equal sign used to assign a path to the parameter. Let's treat it as an empty path too:

```
% bin/wp import --path some
Error: The --path parameter cannot be empty when it is provided
```

Then handle an empty string as an empty path:

```
% bin/wp import --path=
Error: The --path parameter cannot be empty when it is provided
```

When a non-empty path is provided, include the full working directory in the error message:

```
% bin/wp import --path=some
Error: This (/Volumes/Sites/wp-cli/some/) does not seem to be a WordPress installation.
Pass --path=`path/to/wordpress` or run `wp core download`.
```

Fixes: https://github.com/wp-cli/wp-cli/issues/5559

### CODE REVIEW

**Review by danielbachhuber — COMMENTED** [review]
Thanks @wojtekn !

Can you write some [functional tests](https://make.wordpress.org/cli/handbook/contributions/pull-requests/#functional-tests) for this change?

[The handbook is editable](https://github.com/wp-cli/handbook/blob/main/pull-requests.md) if there are improvements you identify.

**Comment by wojtekn** [comment]
@danielbachhuber that is good idea, I've added functional tests for three cases.

File: php/WP_CLI/Runner.php [hunk 1]
```diff
@@ -966,7 +972,8 @@ private function check_wp_version() {
 				);
 			}
 			WP_CLI::error(
-				"This does not seem to be a WordPress installation.\n" .
+				"The path does not seem to be a WordPress installation.\n" .
```
Comment (danielbachhuber) [inline php/WP_CLI/Runner.php:975]:
```suggestion
				"This path does not seem to be a WordPress installation.\n" .
```

If we restore the original string, we can probably avoid breaking/changing a lot of test assertions.

File: php/WP_CLI/Runner.php [hunk 2]
```diff
@@ -244,6 +244,12 @@ private static function extract_subdir_path( $index_path ) {
 	 * @return string An absolute path
 	 */
 	private function find_wp_root() {
+		if ( isset( $this->config['path'] ) &&
+			( is_bool( $this->config['path'] ) || empty( $this->config['path'] ) )
+		) {
+			WP_CLI::error( 'The --path parameter cannot be empty when provided' );
```
Comment (danielbachhuber) [inline php/WP_CLI/Runner.php:250]:
```suggestion
			WP_CLI::error( 'The --path parameter cannot be empty when provided.' );
```

**Review by danielbachhuber — APPROVED** [review]
Looks good to me 👍

**Pull request merged**